src/core/types/decimal192.rs

```diff
@@ -124,19 +149,11 @@ impl TryFrom<f32> for Decimal {
     ///
     /// assert!(Decimal::try_from(208050.17).is_ok());
     ///
-    /// assert_eq!(
-    ///     Decimal::try_from(f32::MIN_POSITIVE),
-    ///     Err(CommonError::DecimalOverflow { bad_value: f32::MIN_POSITIVE.to_string() })
-    /// );
+    /// assert!(Decimal::try_from(f32::MIN_POSITIVE).is_ok());
     /// ```
     fn try_from(value: f32) -> Result<Self, Self::Error> {
```

Comment:
is it now in fact possible to get it to throw from an `f32`? it seems not? Maybe change this `TryFrom<f32>` into a `From<f32>` (return `Decimal192` instead of `Result<>`)? If you ensure to try `MAX_POSITIVE` and `MIN_POSITIVE`, `MAX_NEGATIVE` and `MIN_NEGATIVE` and none of those for 4 scenarios throw, then this cannot throw and it is safe to force unwrap

and if so, also change the `pub fn new_decimal_from_f32(value: f32) -> Result<Decimal192>` to not return `Result`.

What about `f64`? Same thing applies?

Reply:
Good point! This applies to f32, but not to f64 as the parsing into `Scrypto192Decimal` fails with `Overflow` error.